presto-kafka/src/main/java/io/prestosql/plugin/kafka/KafkaSecurityConfig.java

```diff
@@ -0,0 +1,142 @@
+import org.apache.kafka.common.config.SslConfigs;
+
+import java.util.Locale;
+import java.util.Properties;
+
+/**
+ *  {@KafkaSecurityConfig} manages Kafka configuration related to connection security. I.e. if the security protocol
+ *  'SSL' is being used. If so, additional properties might be supplied, i.e. location and password of the truststore
+ *  and keystore to be used.
+ */
+public class KafkaSecurityConfig
+{
+    private Properties props;
+
+    private String sslKeystoreFile;
+    private String sslKeystorePassword;
+    private String sslKeyPassword;
+    private String sslTruststoreFile;
+    private String sslTruststorePassword;
+    private KafkaEndpointIdentificationAlgorithm sslEndpointIdentificationAlgorithm;
+
+    @Config("kafka.keystore-file")
+    @LegacyConfig("kafka.ssl.keystore.location")
+    @ConfigDescription(SslConfigs.SSL_KEYSTORE_LOCATION_DOC)
+    public void setSslKeystoreFile(String sslKeystoreFile)
+    {
+        this.sslKeystoreFile = sslKeystoreFile;
+    }
+
+    @FileExists
+    public String getSslKeystoreFile()
+    {
+        return sslKeystoreFile;
+    }
+
+    @Config("kafka.keystore-password")
+    @LegacyConfig("kafka.ssl.keystore.password")
+    @ConfigDescription(SslConfigs.SSL_KEYSTORE_PASSWORD_DOC)
+    @ConfigSecuritySensitive
+    public void setSslKeystorePassword(String sslKeystorePassword)
+    {
+        this.sslKeystorePassword = sslKeystorePassword;
+    }
+
+    public String getSslKeystorePassword()
+    {
+        return sslKeystorePassword;
+    }
+
+    @Config("kafka.key-password")
+    @LegacyConfig("kafka.ssl.key.password")
+    @ConfigDescription(SslConfigs.SSL_KEY_PASSWORD_DOC)
+    @ConfigSecuritySensitive
+    public void setSslKeyPassword(String sslKeyPassword)
+    {
+        this.sslKeyPassword = sslKeyPassword;
+    }
+
+    public String getSslKeyPassword()
+    {
+        return sslKeyPassword;
+    }
+
+    @Config("kafka.truststore-file")
+    @LegacyConfig("kafka.ssl.truststore.location")
+    @ConfigDescription(SslConfigs.SSL_TRUSTSTORE_LOCATION_DOC)
+    public void setSslTruststoreFile(String sslTruststoreFile)
+    {
+        this.sslTruststoreFile = sslTruststoreFile;
+    }
+
+    @FileExists
+    public String getSslTruststoreFile()
+    {
+        return sslTruststoreFile;
+    }
+
+    @Config("kafka.truststore-password")
+    @LegacyConfig("kafka.ssl.truststore.password")
+    @ConfigDescription(SslConfigs.SSL_TRUSTSTORE_PASSWORD_DOC)
+    @ConfigSecuritySensitive
+    public void setSslTruststorePassword(String sslTruststorePassword)
+    {
+        this.sslTruststorePassword = sslTruststorePassword;
+    }
+
+    public String getSslTruststorePassword()
+    {
+        return sslTruststorePassword;
+    }
+
+    @Config("kafka.endpoint-identification-algorithm")
+    @ConfigDescription(SslConfigs.SSL_ENDPOINT_IDENTIFICATION_ALGORITHM_DOC)
+    public void setSslEndpointIdentificationAlgorithm(String sslEndpointIdentificationAlgorithm)
```

Comment:
```suggestion
    public void setSslEndpointIdentificationAlgorithm(KafkaEndpointIdentificationAlgorithm sslEndpointIdentificationAlgorithm)
```

airlift handles enums internally

Reply:
done